tox.ini

```diff
@@ -37,3 +38,17 @@ commands =
 deps =
     Sphinx
     pylons-sphinx-themes
+
+
+[testenv:lint]
+skip_install = true
+commands =
+    check-manifest
+    # build sdist/wheel
+    python setup.py sdist
+    python setup.py bdist_wheel
+    twine check dist/*
+deps =
+    check-manifest
+    readme_renderer
+    twine
```

Comment:
This looks more like a build than a lint.

[`readme_renderer` should be installed when you install `twine`](https://stackoverflow.com/questions/46682793/how-to-ensure-that-readme-rst-is-valid/46687472#46687472), so it is not necessary to include it separately.

I would suggest the following, adapted from [Pyramid](https://github.com/Pylons/pyramid/blob/master/tox.ini#L72-L84).

```
[testenv:build]
skip_install = true
commands =
    # clean up build/ and dist/ folders
    python -c 'import shutil; shutil.rmtree("dist", ignore_errors=True)'
    python setup.py clean --all
    # build sdist
    python setup.py sdist --dist-dir {toxinidir}/dist
    # build wheel from sdist
    pip wheel -v --no-deps --no-index --no-build-isolation --wheel-dir {toxinidir}/dist --find-links {toxinidir}/dist zodburi
    check-manifest
    twine check dist/*
deps =
    check-manifest
    setuptools
    twine
    wheel
```

Using a couple of other Pylons Project projects as examples, you could add a `[testenv:lint]` that uses black, isort, and optionally more. In Deform, we put the [lint deps in setup.py](http://github.com/Pylons/deform/blob/master/setup.py#L41-L49), then refer to that in [tox.ini](https://github.com/Pylons/deform/blob/main/tox.ini#L22-L32). Pyramid uses only isort and black in a `format` block. It's up to you what you want to do here as a maintainer.

Reply:
@stevepiercy, thanks for the pointers. I've removed `readme_renderer` and split lint into lint + build. For now I prefer to keep lint minimal with only check-manifest.